### PR TITLE
Ik 67

### PR DESCRIPTION
**Pull request by handriamanantena**
This branch is related to receiving notifications. Sending notifications will be done in a separate branch.
Notifications can be sent through the firebase console.
If you want to test it out...

Go to the following link https://firebase.google.com/docs/notifications/android/console-device
Follow the steps in the section "Add Firebase to your app".
Here they explain how to connect firebase to the application and then send a notification the device.
sending a notification is done through their console.

### CODE REVIEW

**Review by arosser — COMMENTED** [review]
.

File: client/app/src/main/java/com/example/mrides/Notification/FirebaseInstanceIdService.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package com.example.mrides.Notification;
+
+/**
+ * Created by Deez Nuts on 3/10/2017.
```
Comment (arosser) [inline client/app/src/main/java/com/example/mrides/Notification/FirebaseInstanceIdService.java]: Unprofessionalism

File: client/app/src/main/java/com/example/mrides/Activity/MainActivity.java [hunk 2]
```diff
@@ -167,19 +172,32 @@ public void onComplete(@NonNull Task<AuthResult> task) {
...
+     *
+     * @param connectionResult
+     */
     @Override
     public void onConnectionFailed(@NonNull ConnectionResult connectionResult) {
 
     }
 
+    /**
+     * This method handles loging in and logging out the user.
```
Comment (arosser) [inline client/app/src/main/java/com/example/mrides/Activity/MainActivity.java]: typo on logging in

**Review by arosser — CHANGES_REQUESTED** [review]

File: client/app/src/main/java/com/example/mrides/Activity/MainActivity.java [hunk 3]
```diff
@@ -107,7 +107,7 @@ public void onConnectionFailed(@NonNull ConnectionResult connectionResult) {
 
     @Override
     public void responseReceived(String response) {
-
+        System.out.print(response);
```
Comment (redfox300) [inline client/app/src/main/java/com/example/mrides/Activity/MainActivity.java]: If finished debugging, should remove these type of print statements.

File: client/app/src/main/java/com/example/mrides/Activity/TempMainActivity.java [hunk 4]
```diff
@@ -53,7 +54,9 @@ protected void onCreate(Bundle savedInstanceState) {
     }
 
     public void signOut(View view) {
-
+        FirebaseAuth.getInstance().signOut();
+        //Intent intent = new Intent(TempMainActivity.this, MainActivity.class);
```
Comment (redfox300) [inline client/app/src/main/java/com/example/mrides/Activity/TempMainActivity.java]: To remove

File: client/app/src/main/java/com/example/mrides/Activity/MainActivity.java [hunk 5]
```diff
@@ -50,43 +49,39 @@ protected void onCreate(Bundle savedInstanceState) {
                 .requestIdToken(getString(R.string.default_web_client_id))
                 .requestEmail()
                 .build();
-        // Build a GoogleApiClient with access to the Google Sign-In API and the
-        // options specified by gso.
         mGoogleApiClient = new GoogleApiClient.Builder(this)
                 .enableAutoManage(this /* FragmentActivity */, (GoogleApiClient.OnConnectionFailedListener) this /* OnConnectionFailedListener */)
```
Comment (redfox300) [inline client/app/src/main/java/com/example/mrides/Activity/MainActivity.java]: Remove commented code.

Comment (redfox300) [inline client/app/src/main/java/com/example/mrides/Activity/MainActivity.java:60]: Remove commented code as it adds confusion.

File: client/app/src/main/java/com/example/mrides/Activity/MainActivity.java [hunk 6]
```diff
@@ -165,4 +166,8 @@ public void responseReceived(String response) {
         startActivity(intent);
     }
 
+    @Override
+    public void onConnectionFailed(@NonNull ConnectionResult connectionResult) {
```
Comment (redfox300) [inline client/app/src/main/java/com/example/mrides/Activity/MainActivity.java]: Empty method.

File: client/app/src/main/java/com/example/mrides/Activity/ActivityObserver.java [hunk 7]
```diff
@@ -1,8 +1,14 @@
 package com.example.mrides.Activity;
 
 
-
 public interface ActivityObserver {
 
+    /**
```
Comment (redfox300) [inline client/app/src/main/java/com/example/mrides/Activity/ActivityObserver.java:6]: Good addition of javadoc, this should be added to at least the critical methods in your code.

**Review by arosser — APPROVED** [review]

**Pull request merged**